discord/commands/options.py

```diff
@@ -285,6 +285,20 @@ def __init__(
                 for o in kwargs.pop("choices", [])
             ]
 
+        # this is done here so it does not have to be computed every time the autocomplete is invoked
+        if self.autocomplete:
```

Comment:
This is bad bc `Option.autocomplete` is documented, so advanced users might for some reason that I cannot understand change it at runtime - which would lead to AttributeError when internally accessing `autocomplete._is_instance_method`.

Reply:
Then why not make it a property so it is handled on the setter?

Reply:
That was my idea yeah, but I don't have time rn, added this comment to remind me to do it

Reply:
I just wanted to make a quick demo so Debauchery could test, and work on it later in more detail

Reply:
Could also move some logic to utils - smth like `count_required_params` or idk